### PR TITLE
fix: intersects with v0.0.0 and v0.0.0-0

### PR DESCRIPTION
**Pull request by wraithgar**
Also fix options not being passed through from semver.intersects

Closes https://github.com/npm/node-semver/issues/521

### CODE REVIEW

**Comment by wraithgar** [comment]
This can't land before #535 since that removes the dead code that is causing coverage failures here.

File: classes/comparator.js [hunk 1]
```diff
@@ -97,6 +97,16 @@ class Comparator {
...
+      return false
+    }
+    if (!options.includePrerelease &&
+      (this.value.startsWith('<0.0.0') || comp.value.startsWith('<0.0.0'))) {
+      return false
+    }
+
     const sameDirectionIncreasing =
       (this.operator === '>=' || this.operator === '>') &&
       (comp.operator === '>=' || comp.operator === '>')
```
Comment (H4ad) [inline classes/comparator.js:112]: It is not worth adding in this PR also the early returns mentioned here: https://github.com/npm/node-semver/pull/528#issuecomment-1499241387?

Comment (wraithgar) [reply classes/comparator.js:112]: Sure, once I've gotten any feedback on the implementation of the bugfix we can add that as a separate commit.

Comment (wraithgar) [reply classes/comparator.js]: Much cleaner now.

**Pull request merged**